### PR TITLE
Refresh token fix

### PR DESCRIPTION
**Pull request by Acorn221**
I'm not sure what I did or if this is different now, but I think I confused the refresh token with the authorisation code?

[specifically in this commit, I'm not sure if it has always been like this](https://github.com/alecchendev/spotify-refresh-token/commit/bfc40d61e18fd3c46e4261c3bff99ddbf68c8a74)

### CODE REVIEW

**Comment by alecchendev** [comment]
Hey, thanks for the PR! Couple requests to help me review:
- Can you squash your commits so this PR doesn't include previous commits in the main branch's history (and just shows this change as commits to be added on top of main as it is currently)?
  - (Also I'd recommend rebasing on main instead of merging main into your branch to get the updated state)
- Can you elaborate a little on what this change is supposed to be - is it mainly naming or an actual logic change? Etc.

**Comment by Acorn221** [comment]
Hi, sorry I should have been more clear. I think when I made my initial PR to this repo, I messed up and provided the wrong key to the user in the "refreshToken" area [here's the updated code deployed on my branch](https://acorn221.github.io/spotify-token-getter/). The token provided seems to be a lot longer than it should be in the refreshToken area and I kept getting incorrect token errors with the spotify API.

I'll go squash the commits too.

**Comment by ReubenFrankel** [comment]
Could have sworn I tested everything E2E when I did my PR a while back, but I am running into this error as well now... When trying to generate a new access token with the refresh token obtained from the app, I get:

```json
{
    "error": "invalid_grant",
    "error_description": "Invalid refresh token"
}
```

**Comment by Acorn221** [comment]
Yeah it seems strange, I thought it was was all working too.
This PR should be fine to merge in, I've squashed my commits and sorted out my branch a little.
I think the issue was just that I confused the oauth2 code with the refresh token when I created the initial big refactor.

**Review by alecchendev — APPROVED** [review]
LGTM

File: src/App.jsx [hunk 1]
```diff
@@ -31,17 +31,17 @@ const allScopes = [
 let callbackUri = window.location.href.split('/').slice(0, 4).join('/');
 
 // if the callback uri ends with a slash, remove it
-callbackUri = callbackUri.charAt(callbackUri.length - 1) === '/' ? callbackUri.slice(0, callbackUri.length - 1) : callbackUri;
+callbackUri = callbackUri.endsWith() === '/' ? callbackUri.slice(0, callbackUri.length - 1) : callbackUri;
```
Comment (ReubenFrankel) [inline src/App.jsx:34]: Theres a small bug here - should be:

```
callbackUri = callbackUri.endsWith('/') ? callbackUri.slice(0, callbackUri.length - 1) : callbackUri;
```

Comment (ReubenFrankel) [reply src/App.jsx:34]: Fixed here: https://github.com/alecchendev/spotify-refresh-token/pull/9/commits/9986b59ecc9bc1f3ca52dd269446b6f97b79424f

**Pull request merged**